components/script/dom/navigator.rs

```diff
@@ -81,9 +82,52 @@ impl Navigator {
         self.xr.get()
     }
 
-    pub fn gamepads(&self) -> DomRoot<GamepadList> {
-        self.gamepads
-            .or_init(|| GamepadList::new(&self.global(), &[]))
+    pub fn get_gamepad(&self, index: usize) -> Option<DomRoot<Gamepad>> {
+        self.gamepads.borrow().get(index).and_then(|g| g.get())
+    }
+
+    pub fn set_gamepad(&self, index: usize, gamepad: Option<DomRoot<Gamepad>>) {
```

Comment:
I think it's a good idea to:

1. pass a `&Gamepad` instead of the root as argument.
2. Break this in two funtions, one `set_gamepad` and another `remove_gamepad`, that way the logic becomes easier to follow. We can then also move the call to `shrink_gamepads_list`(which can be private then) into `remove_gamepad`. Sorry didn't think about this until I saw it...